exporter/signalfxexporter/dpclient.go

```diff
@@ -75,32 +74,28 @@ func (s *sfxDPClient) pushMetricsData(
 	ctx context.Context,
 	md pdata.Metrics,
 ) (droppedDataPoints int, err error) {
-	var numDroppedDataPoints int
-	var errs []error
-
 	rms := md.ResourceMetrics()
+	if rms.Len() == 0 {
+		return 0, nil
+	}
+
+	// All metrics in the pdata.Metrics will have the same access token because of the BatchPerResourceMetrics.
+	metricToken := s.retrieveAccessToken(rms.At(0))
```

Comment:
If access token passthrough is enabled, the token is set as a resource attribute for each resource metric https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/receiver/signalfxreceiver/receiver.go#L267 to allow functioning w/ the initial batching logic.  It appears we are no longer deleting these for each one so they will leak w/ the datapoint.

Reply:
Not true, see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1727/files#diff-c929f651c8316972be7c0a8d184c4d0756c7b19d1e9cd0a087970a90282b7bf6R431

Reply:
nice, missed this

Reply:
Marking as done.